### PR TITLE
Inventory copy

### PR DESCRIPTION
**Pull request by d-chambers**
### What does this PR do?

Adds a copy method to `Inventory` to be consistent with `Catalog` and `Stream` which both have copy methods that created deep copies.

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [x] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [x] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by megies** [comment]
Rebased on current master and force-pushed so that we have fresh CI results tomorrow.

File: obspy/core/inventory/inventory.py [hunk 1]
```diff
@@ -320,6 +320,33 @@ def write(self, path_or_file_object, format, **kwargs):
...
+        :rtype: :class:`~obspy.core.inventory.inventory.Inventory`
+        :return: Copy of current inventory.
+
+        .. rubric:: Examples
+
+        1. Create an Inventory and copy it
+
+            >>> from obspy import read_inventory
+            >>> inv = read_inventory()
+            >>> inv2 = inv2.copy()
```
Comment (ThomasLecocq) [inline obspy/core/inventory/inventory.py]: inv2 = inv.copy()

File: obspy/core/tests/test_inventory.py [hunk 2]
```diff
@@ -574,6 +574,19 @@ def test_inventory_can_be_initialized_with_no_arguments(self):
...
+        """
+        Test for copying inventory.
+        """
+        inv = read_inventory()
+        inv2 = inv.copy()
+        self.assertIsNot(inv, inv2)
+        self.assertEqual(inv, inv2)
+        # make sure changing inv2 doesnt affect inv
+        original_latitude = inv2[0][0][0].latitude
+        inv2[0][0][0].latitudue = original_latitude + 1
```
Comment (ThomasLecocq) [inline obspy/core/tests/test_inventory.py]: latitudue  - latitude

Comment (ThomasLecocq) [reply obspy/core/tests/test_inventory.py]: not sure how to understand this test...

Comment (Jollyfant) [reply obspy/core/tests/test_inventory.py]: It just checks whether the memory copy works by modifying the copied reference and checking it doesn't touch the original reference.

Comment (megies) [reply obspy/core/tests/test_inventory.py]: um.. but the latitude is never actually changed anywhere.. so i agree with @ThomasLecocq

Comment (Jollyfant) [reply obspy/core/tests/test_inventory.py]: I mean the typo has to be fixed.. but the point is to prove that the two objects don't reference the same point in memory

Comment (megies) [reply obspy/core/tests/test_inventory.py]: Fixed the typo already and also amended the test

**Comment by megies** [comment]
CI all green so far, only 1-2 appveyor jobs not yet done

**Comment by ThomasLecocq** [comment]
@megies all green (except unrelated evalresp)

**Comment by megies** [comment]
Bääääm! :rocket:

**Pull request merged**